src/components/cards/BaseStakingCard.tsx

```diff
@@ -179,20 +180,19 @@ const BaseStakingCard: FC<IBaseStakingCardProps> = ({
 				: config.MAINNET_NETWORK_NUMBER;
 		const checkNetworkAndShowStakeModal = async () => {
 			if (_chain && _open === type) {
-				await switchNetwork(_chainId);
+				dispatch(setShowWalletModal(!isWalletActive));
+				if (isWalletActive) await switchNetwork(_chainId);
```

Comment:
It calls the switchNetwork function immediately after showing the wallet modal. so I think It will be buggy.

Reply:
will `isWalletActive` be positive?

Reply:
Yeah... you are right ;)